### PR TITLE
 #337/3/UI/Desktop/Sidebar - clean

### PR DESCRIPTION
**Pull request by RobertS-ki**
Added Sidebar on the left side of the page:
obraz

### CODE REVIEW

File: client/src/Atoms/NavLinkStyled/NavLinkStyled.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import styled from 'styled-components';
+import { NavLink } from 'react-router-dom';
+
+const bgHandler = (props) => {
+  if (props.stylefor === 'Sidebar')
+    return `background: ${props.theme.primary101}`;
+  if (props.stylefor === 'Cornermenu')
+    return `color: ${props.theme.grey400}; font-weight: normal`;
+  if (props.stylefor === 'Navbar') return `color: ${props.theme.primary801}`;
```
Comment (vieraboschkova) [inline client/src/Atoms/NavLinkStyled/NavLinkStyled.js:9]: Perhaps you could add a default case? if none of the above -> then render stg

File: client/src/Molecules/MenuItem/MenuItem.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        {text}
+      </NavLinkStyled>
+    </MenuItemStyled>
+  );
+};
+MenuItem.propTypes = {
+  text: propTypes.string,
+  linkTo: propTypes.string,
+  icon: propTypes.object,
+  stylefor: propTypes.string,
```
Comment (vieraboschkova) [inline client/src/Molecules/MenuItem/MenuItem.js:22]: Any of them should be required maybe?

File: client/src/Organisms/BurgerMenu/BurgerMenuStyled.js [hunk 3]
```diff
@@ -25,9 +25,10 @@ const BurgerMenuStyled = styled.div`
     grid-area: 3 / 1 / 4 / 2;
     transform: none;
     transition: none;
+    display: none;
   }
   @media only screen and (min-width: 1600px) {
-    grid-area: 3 / 2 / 4 / 3;
+    display: none;
```
Comment (vieraboschkova) [inline client/src/Organisms/BurgerMenu/BurgerMenuStyled.js:31]: This media query is not neccessary anymore. If you add display none from 800 up, you dont need anythin else.

File: client/src/Organisms/BurgerMenu/BurgerMenuStyled.js [hunk 4]
```diff
@@ -25,9 +25,10 @@ const BurgerMenuStyled = styled.div`
     grid-area: 3 / 1 / 4 / 2;
```
Comment (vieraboschkova) [inline client/src/Organisms/BurgerMenu/BurgerMenuStyled.js:25]: If display is none, you dont need any other css, it wont be there anywya, they can be removed

File: client/src/Organisms/Sidebar/Sidebar.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import SidebarStyled from './SidebarStyled';
+import { UserDataContext } from './../../Context/UserDataContext';
+
+const Sidebar = () => {
+  const { state } = useContext(UserDataContext);
+  const { isAuthenticated } = state;
+  const { userId } = state;
+  const fontsize = '30px';
+
+  if (isAuthenticated) {
```
Comment (vieraboschkova) [inline client/src/Organisms/Sidebar/Sidebar.js:16]: You could do "early return" pattern here, so switch the order. If(!isAuthenticated?){ render null} else {rest of it} or you could try to write return ({isAuthenticated? (all stuff to render):null}). Just sother solutions, no need to refactor

File: client/src/Organisms/Sidebar/SidebarStyled.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  border-top: none;
+  border-bottom: none;
+  text-align: left;
+
+  @media only screen and (min-width: 800px) {
+    display: grid;
+    grid-area: 3 / 1 / 4 / 2;
+  }
+  @media only screen and (min-width: 1600px) {
+    display: grid;
```
Comment (vieraboschkova) [inline client/src/Organisms/Sidebar/SidebarStyled.js:15]: If you set the display to grid from 800px up, no need to set it again

File: client/src/Templates/Layout/Grid.js [hunk 7]
```diff
@@ -14,7 +14,8 @@ const Grid = styled.div`
     grid-template-columns: 220px 1fr;
   }
   @media only screen and (min-width: 1600px) {
-    grid-template-columns: 1fr 220px minmax(360px, 1160px) 220px 1fr;
+    // 230px is minimal value for the column width for the last sidebar menu item - is it acceptable for the main content window?
+    grid-template-columns: 1fr 230px minmax(360px, 1160px) 220px 1fr;
```
Comment (vieraboschkova) [inline client/src/Templates/Layout/Grid.js:18]: Perhaps makes the sides the same size?

**Comment by vieraboschkova** [comment]
image
The top navigation is missing, is this a bug or is stg you are working on?

**Comment by vieraboschkova** [comment]
image
Do you need the left border? If not remove it, also set the width of border it in px not rem, and perhaps check with @pierwszazlewej  where should it end (probably something to check once top nav is back)

**Comment by vieraboschkova** [comment]
Perhaps adjust the settings for the grid, the side menu seems really tight in comparison to the rest, may be best to set the minimal value.

**Comment by vieraboschkova** [comment]
Remember to update build so we can view it in vercel. Good job!

File: client/src/Templates/Layout/Grid.js (hunk 7 above)
Comment (RobertS-ki) [reply client/src/Templates/Layout/Grid.js:18]: That will be discussed with the team. See the question in the comment above.

File: client/src/Organisms/Sidebar/SidebarStyled.js (hunk 6 above)
Comment (RobertS-ki) [reply client/src/Organisms/Sidebar/SidebarStyled.js:15]: Fixed.

File: client/src/Organisms/Sidebar/Sidebar.js (hunk 5 above)
Comment (RobertS-ki) [reply client/src/Organisms/Sidebar/Sidebar.js:16]: Interesting...

File: client/src/Organisms/BurgerMenu/BurgerMenuStyled.js (hunk 3 above)
Comment (RobertS-ki) [reply client/src/Organisms/BurgerMenu/BurgerMenuStyled.js:31]: True. Fixed.

File: client/src/Organisms/BurgerMenu/BurgerMenuStyled.js (hunk 4 above)
Comment (RobertS-ki) [reply client/src/Organisms/BurgerMenu/BurgerMenuStyled.js:25]: Fixed.

File: client/src/Molecules/MenuItem/MenuItem.js (hunk 2 above)
Comment (RobertS-ki) [reply client/src/Molecules/MenuItem/MenuItem.js:22]: text isRequired. Fixed.

File: client/src/Atoms/NavLinkStyled/NavLinkStyled.js (hunk 1 above)
Comment (RobertS-ki) [reply client/src/Atoms/NavLinkStyled/NavLinkStyled.js:9]: If non of te above, css style reverts to what is defined outside the handler function. That is for the stylefor='Sidebar": no left padding and no background. I will clarify if this is 100% ok in the Navbar PR - which is also to be reviewed. See you there!

**Review by n0macx — APPROVED** [review]
Build is great the rest... also great

**Pull request merged**